tket/src/Predicates/PassGenerators.cpp

```diff
@@ -751,4 +752,20 @@ PassPtr GlobalisePhasedX(bool squash) {
   return std::make_shared<StandardPass>(precons, t, postcon, j);
 }
 
+PassPtr CustomPass(
+    std::function<Circuit(const Circuit&)> transform,
+    const std::string& label) {
+  Transform t{[transform](Circuit& circ) {
+    Circuit circ_in = circ;
+    circ = transform(circ);
+    return circ != circ_in;
```

Comment:
not sure what the performance implications are but this looks more natural to me, it might avoid a copy?
```suggestion
    Circuit circ_out = transform(circ);
    bool success circ != circ_out;
    circ = circ_out;
    return success
```